test/transpiler/transpiler_binary_operator_test.ts

```diff
@@ -9,127 +10,142 @@ describe('Transpiler', () => {
+            const wrapper = transpiler.transpile('function sub(a, b) { return a - b; }', type);
             wrapper.setFunctionName('sub');
 
             expect(wrapper.call(1, 2)).to.equal(-1);
             expect(wrapper.call(10, 2)).to.equal(8);
             expect(wrapper.call(-20, 20)).to.equal(-40);
-            expect(wrapper.call(NaN, 2)).to.equal(-2);
         });
 
         it('should handle multiplication', () => {
-            const wrapper = transpiler.transpile('function mul(a, b) { return a * b }');
+            const type = createFunctionWithIntParameters('mul', 2);
+            const wrapper = transpiler.transpile('function mul(a, b) { return a * b }', type);
             wrapper.setFunctionName('mul');
 
             expect(wrapper.call(1, 2)).to.equal(2);
             expect(wrapper.call(10, 2)).to.equal(20);
             expect(wrapper.call(-10, 2)).to.equal(-20);
-            expect(wrapper.call(NaN, 2)).to.equal(0);
         });
 
         it('should handle multiplication before addition', () => {
-            const wrapper = transpiler.transpile('function mul(a, b, c) { return a + b * c }');
+            const type = createFunctionWithIntParameters('mul', 3);
+            const wrapper = transpiler.transpile('function mul(a, b, c) { return a + b * c }', type);
 
             expect(wrapper.setFunctionName('mul').call(3, 2, 5)).to.equal(13);
         });
 
         it('should handle multiplication before subtraction', () => {
-            const wrapper = transpiler.transpile('function mul(a, b, c) { return a - b * c }');
+            const type = createFunctionWithIntParameters('mul', 3);
+            const wrapper = transpiler.transpile('function mul(a, b, c) { return a - b * c }', type);
 
             expect(wrapper.setFunctionName('mul').call(3, 2, 5)).to.equal(-7);
         });
 
         it('should handle division', () => {
-            const wrapper = transpiler.transpile('function div(a, b) { return a / b }');
+            const type = createFunctionWithIntParameters('div', 2);
+            const wrapper = transpiler.transpile('function div(a, b) { return a / b }', type);
             wrapper.setFunctionName('div');
 
             expect(wrapper.call(1, 2)).to.equal(0);
             expect(wrapper.call(10, 2)).to.equal(5);
             expect(wrapper.call(-10, 2)).to.equal(-5);
-            expect(wrapper.call(NaN, 2)).to.equal(0);
         });
 
         it('should handle division by 0', () => {
-            const wrapper = transpiler.transpile('function div(a, b) { return a / b }');
+            const type = createFunctionWithIntParameters('div', 2);
+            const wrapper = transpiler.transpile('function div(a, b) { return a / b }', type);
             wrapper.setFunctionName('div');
 
             expect(() => wrapper.call(2, 0)).to.throw();
-            expect(() => wrapper.call(2, NaN)).to.throw();
+            expect(() => wrapper.call(10, 0)).to.throw();
```

Comment:
This test is the same as the one above => remove it.

Reply:
I added this test because it could be that just `2 / 0` throws an error. Maybe a negative number would be better?

Reply:
It doesn't matter that much. Let's just leave it as it is now.